File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneOneInteger.java
package com.hartwig.actin.treatment.input.single;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class })
public abstract class OneGeneOneInteger {

    @NotNull
    public abstract String geneName();

    public abstract int integer();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneOneInteger.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class])
abstract class OneGeneOneInteger() {
    abstract fun geneName(): String
    abstract fun integer(): Int
}
